src/main/java/org/openstreetmap/atlas/generator/AtlasGenerator.java

```diff
@@ -295,222 +392,266 @@ public void start(final CommandMap command)
-                    }
-                    if (pbfWayConfiguration != null)
-                    {
-                        atlasLoadingOption.setOsmPbfWayFilter(
-                                getTaggableFilterFrom(pbfWayConfiguration, configurationMap));
-                    }
-                    if (pbfRelationConfiguration != null)
-                    {
-                        atlasLoadingOption.setOsmPbfRelationFilter(
-                                getTaggableFilterFrom(pbfRelationConfiguration, configurationMap));
-                    }
-
-                    // Build the appropriate PbfLoader
-                    final PbfLoader loader = new PbfLoader(pbfContext, sparkContext, boundaries,
-                            atlasLoadingOption, codeVersion, dataVersion, task.getAllShards());
-                    final String name = countryName + CountryShard.COUNTRY_SHARD_SEPARATOR
-                            + shard.getName();
-                    final Atlas atlas;
-                    try
-                    {
-                        // Generate the Atlas for this shard
-                        atlas = loader.load(countryName, shard);
-                    }
-                    catch (final Throwable e)
-                    {
-                        throw new CoreException("Building Atlas {} failed!", name, e);
-                    }
+        if (useRawAtlas)
+        {
+            // AtlasLoadingOption isn't serializable, neither is command map. To avoid duplicating
+            // boiler-plate code for creating the AtlasLoadingOption, extract the properties we need
+            // from the command map and pass those around to create the AtlasLoadingOption
+            final Map<String, String> atlasLoadingOptions = extractAtlasLoadingProperties(command);
+
+            // Generate the raw Atlas and filter any null atlases
+            final JavaPairRDD<String, Atlas> countryRawAtlasShardsRDD = getContext()
+                    .parallelize(tasks, tasks.size()).mapToPair(this.generateRawAtlas(boundaries,
+                            sparkContext, atlasLoadingOptions, pbfContext, atlasScheme))
+                    .filter(tuple -> tuple._2() != null);
+
+            // Persist the RDD and save the intermediary state
+            countryRawAtlasShardsRDD.cache();
+            countryRawAtlasShardsRDD.saveAsHadoopFile(
+                    getAlternateSubFolderOutput(output, RAW_ATLAS_FOLDER), Text.class, Atlas.class,
+                    MultipleAtlasOutputFormat.class, new JobConf(configuration()));
+            logger.info("\n\n********** SAVED THE RAW ATLAS **********\n");
+
+            // Slice the raw Atlas and filter any null atlases
+            final JavaPairRDD<String, Atlas> countrySlicedRawAtlasShardsRDD = countryRawAtlasShardsRDD
+                    .mapToPair(this.sliceRawAtlas(boundaries)).filter(tuple -> tuple._2() != null);
```

Comment:
Here passing the boundaries means the driver will have to serialize it and pass it around to all the slaves. I guess the old flow does the same thing, not sure if there is a better way to do that... Would it be possible to only pass the boundary of the country that is being sliced? Not sure.

Reply:
That's a good point, let me think of a better way to handle this or at least create an issue!

Reply:
Created: https://github.com/osmlab/atlas-generator/issues/28 - let's come back to this. I think it's important, but something we can live with in the meantime.